clients/src/test/java/org/apache/kafka/clients/admin/KafkaAdminClientTest.java

```diff
@@ -376,6 +378,41 @@ public void testDescribeConfigs() throws Exception {
         }
     }
 
+    public void testAlterPartitionCounts() throws Exception {
+        try (MockKafkaAdminClientEnv env = mockClientEnv()) {
+            env.kafkaClient().setNodeApiVersions(NodeApiVersions.create());
+            env.kafkaClient().prepareMetadataUpdate(env.cluster(), Collections.<String>emptySet());
+            env.kafkaClient().setNode(env.cluster().controller());
+
+            Map<String, ApiError> m = new HashMap<>();
+            m.put("my_topic", ApiError.NONE);
+            m.put("other_topic", ApiError.fromThrowable(new InvalidTopicException("some detailed reason")));
+
+            // Test a call where one filter has an error.
+            env.kafkaClient().prepareResponse(new CreatePartitionsResponse(0, m));
+
+            Map<String, NewPartition> counts = new HashMap<>();
+            counts.put("my_topic", NewPartition.increaseTo(3));
+            counts.put("other_topic", NewPartition.increaseTo(3, asList(asList(2), asList(3))));
+
+            CreatePartitionsResult results = env.adminClient().createPartitions(counts);
+            Map<String, KafkaFuture<Void>> values = results.values();
+            KafkaFuture<Void> myTopicResult = values.get("my_topic");
+            assertTrue(!myTopicResult.isCompletedExceptionally());
+            assertEquals(null, myTopicResult.get());
+            KafkaFuture<Void> otherTopicResult = values.get("other_topic");
+            //assertTrue(otherTopicResult.isCompletedExceptionally());
```

Comment:
I removed this assertion. The problem was that it would fail because the future hadn't completed (exceptionally or not): IWO `isCompletedExceptionally()` doesn't wait if the future isn't completed yet. The `get()` call throwing or not is a sufficient test of the functionality I'm adding here.